backend/src/main/java/com/redhat/cloud/notifications/routers/models/Facet.java

```diff
@@ -13,10 +14,25 @@
     @NotNull
     private final String displayName;
 
-    public Facet(@NotNull String id, @NotNull String name, @NotNull String displayName) {
+    private final List<Facet> children;
+
+    // Makes jackson happy (default constructor)
+    private Facet() {
+        this.id = null;
+        this.name = null;
+        this.displayName = null;
+        this.children = null;
+    }
+
+    public Facet(@NotNull String id, @NotNull String name, @NotNull String displayName, List<Facet> children) {
```

Comment:
The `@NotNull` annotations on the class fields should be enough. We don't need the annotations on the constructor args.